lib/Conversion/ExportVerilog/ExportVerilogInternals.h

```diff
@@ -76,6 +80,9 @@ struct GlobalNameTable {
   // This contains prefixes for any typedecl'd enum types. Keys are type-aliases
   // of enum types.
   DenseMap<Type, StringAttr> enumPrefixes;
+
+  /// List of names which are marked as reserved for any name.
+  StringSet<> reservedNames;
```

Comment:
nit: Can we make this `DenseSet<StringAttr>`? It should avoid unnecessary string allocation.